src/main/java/graphql/execution/AbstractAsyncExecutionStrategy.java

```diff
@@ -32,7 +32,7 @@ public AbstractAsyncExecutionStrategy(DataFetcherExceptionHandler dataFetcherExc
                 handleNonNullException(executionContext, overallResult, exception);
                 return;
             }
-            Map<String, Object> resolvedValuesByField = new LinkedHashMap<>();
+            Map<String, Object> resolvedValuesByField = new LinkedHashMap<>(fieldNames.size());
```

Comment:
I know this PR has already been merged and closed, and the issue is perfectly ignorable, but wanted to point out that sizing a map to an _exact_ number of elements is generally a bad idea. With the default load factor of 0.75, sizing a map like this guarantees a need for resizing.

Reply:
feedback is always welcome @kaqqao  :)

The intent of the PR wasn't to avoid resizing at all costs but rather to avoid  wasting memory for `ExecutionResult`. If size is not provided, even for a single element map, `LinkedHashMap` will allocate a 16-elements array (because DEFAULT_INITIAL_CAPACITY=16).

Maybe it is possible to further improve it by requesting loadFactor=1.0?

Reply:
NB: in my project I'm overriding this method to use a custom `ImmutableArrayMap` (no hashing, order is guaranteed by just appending and uniqueness of the keys is guaranteed by GQL).  Why? Because in my project we just send back the `ExecutionResult` as is to the client and in this way we are not wasting memory.